tools/jxlinfo.c

```diff
@@ -386,7 +386,17 @@ static int PrintBasicInfo(FILE* file, int verbose) {
       const size_t remaining = JxlDecoderReleaseBoxBuffer(dec);
       box_size += chunk_size;
       box_index += chunk_size - remaining;
-      box_data = realloc(box_data, box_size);
+      void* temp = realloc(box_data, box_size);
+      if (temp == NULL) {
+        free(box_data);
+        box_data = NULL;
+        box_size = 0;
+        box_index = 0;
+        fprintf(stderr, "Memory reallocation failed\n");
+        break;
+      } else {
+        box_data = temp;
```

Comment:
Another comment I forgot to make: if the true branch is going to break anyway, we don’t need the `else`.

Reply:
good point, done